packages/theme/src/unified/UnifiedTheme.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  Theme as Mui4Theme,
+  ThemeOptions as ThemeOptionsV4,
+} from '@material-ui/core/styles';
+import type { PaletteOptions as PaletteOptionsV4 } from '@material-ui/core/styles/createPalette';
+import { PaletteOptions as PaletteOptionsV5 } from '@mui/material/styles/createPalette';
```

Comment:
```suggestion
import { PaletteOptions as PaletteOptionsV5 } from '@mui/material/styles';
```
At MUI we consider everything that is greater than three level nested import private. Just making sure this won't accidently break over time. For v4 we haven't had this type exported from the root of `@material-ui/core/styles`, which is an overlook from our side.